### PR TITLE
chore(blooms): Expose series/stream metadata when iterating over blooms

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

The bloom block format stores metadata from indexing in the index part of the block (`series` file), which contains information, such as the fields (metadata names) that have been indexed in the bloom filter.

This will be needed in the read path (bloom gateway) later on, to check whether a provided label name from the label filter expression was actually indexed or not.

In this PR, the newly available metadata is used by the bloom inspect tool (located in `./tools/bloom/inspect/`) to list the fields along the series, as well as the individual pages of the series that contain the actual scalable bloom filters.

**Notes for the reviewers**

This PR does not change the contents of the bloom fixtures yet, which still tokenizes log line content into n-grams, see https://github.com/grafana/loki/blob/main/pkg/storage/bloom/v1/test_util.go#L63-L126

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/storage/bloom/v1/builder_test.go [hunk 1]
```diff
@@ -290,7 +283,11 @@ func TestMergeBuilder(t *testing.T) {
 	EqualIterators[*SeriesWithBlooms](
 		t,
 		func(a, b *SeriesWithBlooms) {
-			require.Equal(t, a.Series, b.Series, "expected %+v, got %+v", a, b)
+			require.Equal(t, a.Series.Series, b.Series.Series, "expected series %+v, got %+v", a.Series.Series, b.Series.Series)
+			require.Equal(t, a.Series.Meta.Fields, b.Series.Meta.Fields, "expected fields %+v, got %+v", a.Series.Meta.Fields, b.Series.Meta.Fields)
+			// TODO(chaudum): Investigate why offsets not match
+			// This has not been tested before, so I'm not too worried about something being broken.
+			// require.Equal(t, a.Series.Meta.Offsets, b.Series.Meta.Offsets, "expected offsets %+v, got %+v", a.Series.Meta.Offsets, b.Series.Meta.Offsets)
```
Comment (salvacorts) [inline pkg/storage/bloom/v1/builder_test.go:290]: Can you use `require.ElementsMatch` instead if the order is not important?

**Review by rfratto — COMMENTED** [review]
Didn't see anything on my first pass (other than `.Series.Series` looking odd, but IMO that's something for later). Looks good!

File: pkg/storage/bloom/v1/builder.go [hunk 2]
```diff
@@ -288,6 +289,8 @@ func (mb *MergeBuilder) processNextSeries(
 		chunksToAdd = nextInStore.Chunks.Unless(nextInBlocks.Series.Chunks)
 		chunksCopied += len(nextInStore.Chunks) - len(chunksToAdd)
 		preExistingBlooms = nextInBlocks.Blooms
+		// we also need to carry over existing indexed fields from the series metadata
```
[the remark this answers is not in the document]
Comment (chaudum) [reply pkg/storage/bloom/v1/builder.go:292]: Correct.

**Comment by chaudum** [comment]
> Didn't see anything on my first pass (other than `.Series.Series` looking odd, but IMO that's something for later). Looks good!

Agree. `.SeriesWithMeta.Series` does not make it much better either.

**Review by rfratto — APPROVED** [review]
LGTM since nothing changed since my last review

**Pull request merged**